### PR TITLE
Add function to copy a single pod log in podlogs

### PR DESCRIPTION
**Pull request by jingxu97**
The existing function in podlogs is copyAllLogs from a namespace. Add a
new function to copy log from a single pod.

Change-Id: I39d139a2a57d72a0af3432626b0c02c1b43333d2

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/storage/podlogs/podlogs.go [hunk 1]
```diff
@@ -80,7 +80,21 @@ var expectedErrors = regexp.MustCompile(`container .* in pod .* is (terminated|w
...
+func CopyPodLogs(ctx context.Context, cs clientset.Interface, ns, podName string, to LogOutput) error {
+	options := meta.ListOptions{}
+	if podName != "" {
+		options = meta.ListOptions{
+			FieldSelector: fmt.Sprintf("metadata.name=%s", podName),
+		}
+	}
+	watcher, err := cs.CoreV1().Pods(ns).Watch(context.TODO(), options)
+
+	//watcher, err := cs.CoreV1().Pods(ns).Watch(context.TODO(), meta.ListOptions{})
```
Comment (mattcary) [inline test/e2e/storage/podlogs/podlogs.go:97]: Remove comment

File: test/e2e/storage/podlogs/podlogs.go [hunk 2]
```diff
@@ -80,7 +80,21 @@ var expectedErrors = regexp.MustCompile(`container .* in pod .* is (terminated|w
 // log output of a container was already captured, then capturing does not
 // resume if the pod is marked for deletion.
 func CopyAllLogs(ctx context.Context, cs clientset.Interface, ns string, to LogOutput) error {
-	watcher, err := cs.CoreV1().Pods(ns).Watch(context.TODO(), meta.ListOptions{})
+	return CopyPodLogs(ctx, cs, ns, "", to)
+}
+
+// CopyPodLogs is basically the same as CopyAllLogs execpt that if podName is not empty, it only
```
Comment (mattcary) [inline test/e2e/storage/podlogs/podlogs.go:86]: I'd move the comment from CopyAllLogs to here, changing what's necessary, and then the comment for CopyAllLogs can be just "CopyPodLogs for all current or future pods".

Comment (jingxu97) [reply test/e2e/storage/podlogs/podlogs.go:86]: updated. PTAL

File: test/e2e/storage/podlogs/podlogs.go (hunk 1 above)
Comment (jingxu97) [reply test/e2e/storage/podlogs/podlogs.go:97]: done

File: test/e2e/storage/podlogs/podlogs.go [hunk 3]
```diff
@@ -57,11 +57,18 @@ type LogOutput struct {
 // Matches harmless errors from pkg/kubelet/kubelet_pods.go.
 var expectedErrors = regexp.MustCompile(`container .* in pod .* is (terminated|waiting to start|not available)|the server could not find the requested resource`)
 
-// CopyAllLogs follows the logs of all containers in all pods,
+// CopyPodLogs is basically CopyPodLogs for all current or future pods in the given namespace ns
+func CopyAllLogs(ctx context.Context, cs clientset.Interface, ns string, to LogOutput) error {
```
Comment (pohly) [inline test/e2e/storage/podlogs/podlogs.go:61]: Comment doesn't match function.

File: test/e2e/storage/podlogs/podlogs.go [hunk 4]
```diff
@@ -79,8 +86,15 @@ var expectedErrors = regexp.MustCompile(`container .* in pod .* is (terminated|w
...
 // log output of a container was already captured, then capturing does not
 // resume if the pod is marked for deletion.
-func CopyAllLogs(ctx context.Context, cs clientset.Interface, ns string, to LogOutput) error {
-	watcher, err := cs.CoreV1().Pods(ns).Watch(context.TODO(), meta.ListOptions{})
+func CopyPodLogs(ctx context.Context, cs clientset.Interface, ns, podName string, to LogOutput) error {
+	options := meta.ListOptions{}
+	if podName != "" {
+		options = meta.ListOptions{
+			FieldSelector: fmt.Sprintf("metadata.name=%s", podName),
+		}
```
Comment (pohly) [inline test/e2e/storage/podlogs/podlogs.go:94]: `options.FieldSelectpr = fmt.Sprintf("metadata.name=%s", podName)` would have been shorter.

**Pull request merged**